runtime/vm/ContinuationHelpers.cpp

```diff
@@ -221,6 +221,11 @@ yieldContinuation(J9VMThread *currentThread)
 	currentThread->currentContinuation = NULL;
 	VM_ContinuationHelpers::swapFieldsWithContinuation(currentThread, continuation);
 
+	jboolean finished = J9VMJDKINTERNALVMCONTINUATION_FINISHED(currentThread, continuationObject);
+
+	if (finished) {
+		VM_VMHelpers::setContinuationFinished(continuationStatePtr);
```

Comment:
let's move it even earlier, before swap and reseting currentContinuation
functionally probably the same, but it would be more symmetric with enterContinuation (that sets starts bit after swap and setting currentContinuation)